test/collect.cc

```diff
@@ -1,15 +1,54 @@
 #include "eventuals/collect.h"
 
 #include <set>
+#include <utility>
 #include <vector>
 
 #include "eventuals/iterate.h"
+#include "google/protobuf/repeated_field.h"
 #include "gtest/gtest.h"
 
+template <typename Collection>
```

Comment:
Great to have these! Let's put them in a `protobuf` header and library that we don't add to the default `eventuals` library so others can rely on them via `@eventuals//protobuf` if they want this functionality rather than having to reimplement this again.